Choo rewrite

nicknikolov opened the pull request:
😱😱😱

todo:
- [ ] resize elements (and float them in the future)
- [ ] Revive/rewrite properly `app.log`
- [ ] Dimmed background on modal

vorg commented:
1. Need to merge forward commits with updated pex-renderer
2. Enable source maps with debug: true in browserify call. `server.js:111`
const b = browserify(projectPath + '/index.js', Object.assign({ debug: true }, browserifyInc.args, {
3. App crashes if there is no graphs
SyntaxError (DOM Exception 12): The string did not match the expected pattern.`index.js:29`
TypeError: undefined is not an object (evaluating 'flatGraph.name')  `Graph.js:230`
4. Rename graph store to project
5. Does choo handle event removal (e.g. in `ui/scene.js` functional component
6. Is whole `ui ` folder dead?
7.  `components/canvas.js` AKA Canvas AKA nodeGui AKA nodesCanvas, `components/scene.sj` AKA sceneContainer
8. Event handlers inside html should go out `components/node-picker.js:97` unless they are one line function call like `ui/html/node-picker.js:116` 
9. Open (⌘+ o) shortcute doesn’t work in Safari, copy past etc works
10. `stores/commands.js` can we listen for ‘*’ and try to catch mispelled commands?
11. TODO lol its 2018 use fetch, LOL indeed
12. Fetching assets names on render `bus.emit('ui:fetchAssetNames')` that’s general problem with constantly redrawing guis
13. Anything we can do to share more between `index.js` and `index.export.js`?
14. Rename `components/Graph` to GraphEditor or maybe Workspace?
15. `graph:showComment` = showCommentEditBox right? Comments are always visible
16. Move everything to events in `config/keyboard`? 
{ key: 'd', command: 'DeleteSelectedNodes' },
{ key: 'd', event: ‘command:deleteSelectedNodes' },
17. Why we fire undo actions manually?`lib/keyboard.js` 
if (mapping.command === 'Undo') app.history.undo()
18. createApp is no longer used (and so is 80% of imports in `index.js`
19. Why we switched to retro editor theme? How do organize design process so we don’t commit-overwrite each others taste preferences
20. Collapsing everything to `Edit`  menu in editor might be too much. Recompile and close should stay top level?

nicknikolov commented:
- [ ] Need to merge forward commits with updated pex-renderer
- [ ] Enable source maps with debug: true in browserify call. server.js:111
- [ ] App crashes if there is no graphs FIXED
- [ ] Rename graph store to project FIXED
- [ ] Does choo handle event removal NO BUT FIXED
- [ ] Is whole ui folder dead? YES
- [ ] components/canvas.js AKA Canvas AKA nodeGui AKA nodesCanvas, components/scene.sj AKA sceneContainer
- [ ] Event handlers inside html should go out components/node-picker.js:97 unless they are one line function call like ui/html/node-picker.js:116 AFTER MERGE
- [ ] Open (⌘+ o) shortcute doesn’t work in Safari, copy past etc works AFTER MERGE
- [ ] stores/commands.js can we listen for ‘*’ and try to catch mispelled commands? AFTER MERGE
- [ ] Fetching assets names on render bus.emit('ui:fetchAssetNames') that’s general problem with constantly redrawing guis FIXED
- [ ] Anything we can do to share more between index.js and index.export.js? DON'T KNOW, AFTER MERGE
- [ ] Rename components/Graph to GraphEditor or maybe Workspace? RENAMED TO GRAPH EDITOR
- [ ] graph:showComment = showCommentEditBox right? Comments are always visible YES
- [ ] Move everything to events in config/keyboard? MAYBE, AFTER MERGE
- [ ] Why we fire undo actions manually? UNDO IS NOT A COMMAND
- [ ]createApp is no longer used (and so is 80% of imports in index.js FIXED
- [ ] Why we switched to retro editor theme? How do organize design process so we don’t commit-overwrite each others taste preferences AFTER MERGE
- [ ] Collapsing everything to Edit menu in editor might be too much. Recompile and close should stay top level? AFTER MERGE
